gorgone/modules/centreon/autodiscovery/services/discovery.pm

```diff
@@ -129,34 +130,33 @@ sub send_email {
             next if (defined($contact_send->{$contact_id}));
             $contact_send->{$contact_id} = 1;
 
-            my $message = '';
+            my $body = [];
             foreach my $rule_id2 (keys %{$messages}) {
                 if (defined($self->{discovery}->{rules}->{$rule_id2}->{contact}->{$contact_id})) {
-                    $message .= $messages->{$rule_id2} . '\n';
+                    push @$body, @{$messages->{$rule_id2}};
                 }
             }
 
-            if ($message ne '') {
+            if (scalar(@$body) > 0) {
                 $self->{logger}->writeLogInfo("[autodiscovery] -servicediscovery- $self->{uuid} send email to '" . $contact_id .  "' (" . $self->{discovery}->{rules}->{$rule_id}->{contact}->{$contact_id}->{contact_email} . ")");
 
-                $self->send_internal_action(
-                    action => 'COMMAND',
-                    token => $self->{discovery}->{token} . ':email',
-                    data => {
-                        content => [
-                            {
-                                command => sprintf(
-                                    "/usr/bin/printf '%s' | %s -s '%s' -r %s %s",
-                                    $message,
-                                    $self->{mail_command},
-                                    $self->{mail_subject},
-                                    $self->{mail_from},
-                                    $self->{discovery}->{rules}->{$rule_id}->{contact}->{$contact_id}->{contact_email}
-                                )
-                            }
-                        ]
-                    }
-                );
+                my $smtp = Net::SMTP->new('localhost');
```

Comment:
Shouldn't we add a timeout here as default is 120 (like you said in https://github.com/centreon/centreon-autodiscovery/issues/116) ?

Reply:
It s an issue about command line length. Not about timeout.

Reply:
Ah yes.
Anyway I think we should map the timeout as a config parameter.